Terminal.Gui/View/View.Command.cs

```diff
@@ -0,0 +1,274 @@
+﻿#nullable enable
+using System.ComponentModel;
+
+namespace Terminal.Gui;
+
+public partial class View // Command APIs
+{
+    #region Default Implementation
+
+    /// <summary>
+    ///     Helper to configure all things Command related for a View. Called from the View constructor.
+    /// </summary>
+    private void SetupCommands ()
+    {
+        AddCommand (Command.Accept, RaiseAcceptEvent);
+
+        AddCommand (
+                    Command.HotKey,
+                    () =>
+                    {
+                        SetFocus ();
```

Comment:
Not sure what you mean by "having this here".

By default the HotKey will set focus. 

If a View subclass does not want that, it can call `AddCommand (Command.HotKey, ...)` and do something different. See `RadioGroup`:

```cs
AddCommand (
            Command.HotKey,
            ctx =>
            {
                var item = ctx.KeyBinding?.Context as int?;

                if (HasFocus)
                {
                    if (ctx is { KeyBinding: { } } && (ctx.KeyBinding.Value.BoundView != this || HotKey == ctx.Key?.NoAlt.NoCtrl.NoShift))
                    {
                        // It's this.HotKey OR Another View (Label?) forwarded the hotkey command to us - Act just like `Space` (Select)
                        return InvokeCommand (Command.Select, ctx.Key, ctx.KeyBinding);
                    }
                }

                if (item is { } && item < _radioLabels.Count)
                {
                    if (item.Value == SelectedItem)
                    {
                        return true;
                    }

                    // If a RadioItem.HotKey is pressed we always set the selected item - never SetFocus
                    if (ChangeSelectedItem (item.Value) is null or false)
                    {
                        return true;
                    }

                    return false;
                }

                if (SelectedItem == -1 && ChangeSelectedItem (0) == true)
                {
                    return true;
                }

                SetFocus ();

                return true;
            });
```

For RadioGroup, pressing the hotkey cycles through the radio items WITHOUT focusing the RadioGroup.

Reply:
Sorry I was considering this from the point of view of event subscriber to HotKey event.

If you subscribe to the hot key event and make response handler just be handled true.  You expect there to be no state change? But this line means focus has still changed even though your event handler said it had dealt with it?

Reply:
Here is a unit test

```csharp

    [InlineData(true)]
    [InlineData (false)]
    [Theory]
    [AutoInitShutdown]
    public void HandlingHotKey_ShouldNotChangeFocus (bool handle)
    {
        var v1 = new View (){CanFocus = true};
        var v2 = new View () { CanFocus = true };
        Toplevel top = new ();
        top.Add (v1);
        top.Add (v2);

        Application.Begin (top);

        v1.SetFocus ();

        Assert.True (v1.HasFocus);
        Assert.False (v2.HasFocus);
        v2.HotKey = Key.X.WithAlt;

        bool handlerCalled = false;
        v2.HotKeyCommand += (s, e) =>
                            {
                                e.Handled = handle;
                                handlerCalled = true;
                            };

        Application.Driver?.SendKeys ('X',ConsoleKey.X,false,true,false);

        Assert.True (handlerCalled);

        // If event was handled, there should be no state change i.e. focus should not change to v2 
        Assert.Equal (handle, v2.HasFocus);

        top.Dispose ();
    }
```